diff --git a/src/Modules/CalcActiveSkill.lua b/src/Modules/CalcActiveSkill.lua
index 8526eef..238cf5b 100644
--- a/src/Modules/CalcActiveSkill.lua
+++ b/src/Modules/CalcActiveSkill.lua
@@ -217,6 +217,10 @@ function calcs.createActiveSkill(activeEffect, supportList, env, actor, socketGr
 			-- Track how many active skills are supported by this support effect
 			if supportEffect.isSupporting and activeEffect.srcInstance then
 				supportEffect.isSupporting[activeEffect.srcInstance] = true
+				supportEffect.supportedActiveEffectIds = supportEffect.supportedActiveEffectIds or { }
+				if activeEffect.grantedEffect and activeEffect.grantedEffect.id then
+					supportEffect.supportedActiveEffectIds[activeEffect.grantedEffect.id] = true
+				end
 				if supportEffect.srcInstance ~= activeEffect.srcInstance or not (activeEffect.gemData and activeEffect.gemData.grantedEffect.support) then
 					supportEffect.activeSkillLevel = activeEffect.level
 				end
