diff --git a/src/Modules/CalcSetup.lua b/src/Modules/CalcSetup.lua
index edb3576..05d3418 100644
--- a/src/Modules/CalcSetup.lua
+++ b/src/Modules/CalcSetup.lua
@@ -1983,6 +1983,7 @@ function calcs.initEnv(build, mode, override, specEnv)
 					group = { label = "On Kill Monster Explosion", enabled = true, gemList = { }, source = "Explode", noSupports = true }
 					t_insert(build.skillsTab.socketGroupList, group)
 				end
+				group.noSupports = true
 				-- Hack to remove duplicates
 				local explodeBySource = { }
 				for _, explodeSource in ipairs(env.explodeSources) do
@@ -2011,6 +2012,7 @@ function calcs.initEnv(build, mode, override, specEnv)
 							explodeSource = explodeSource,
 						}
 					end
+					activeGemInstance.noSupports = true
 					t_insert(group.gemList, activeGemInstance)
 				end
 				markList[group] = true
@@ -2052,6 +2054,7 @@ function calcs.initEnv(build, mode, override, specEnv)
 						group = { label = "Thorns", enabled = true, gemList = { }, source = "Thorns", noSupports = true }
 						t_insert(build.skillsTab.socketGroupList, group)
 					end
+					group.noSupports = true
 
 					group.thornsSources = env.thornsSources
 					wipeTable(group.gemList)
@@ -2062,6 +2065,7 @@ function calcs.initEnv(build, mode, override, specEnv)
 						enabled = true,
 						level = 1,
 						triggered = true,
+						noSupports = true,
 					}
 					t_insert(group.gemList, activeGemInstance)
 					markList[group] = true
